// sabor/lib/static.js
(function(){
  var Q, mime, path, fs, cache;
  Q = require('q');
  mime = require('mime');
  path = require('path');
  fs = require('fs');
  cache = {};
  (exports.Static || (exports.Static = {})).file = (function(){
    function file(filename){
      return function(){
        if (filename in cache) {
          return cache[filename];
        }
        return Q.ncall(fs.readFile, fs, filename).then(function(data){
          return cache[filename] = {
            body: [data],
            headers: {
              'content-type': mime.lookup(filename)
            },
            status: 200
          };
        }, function(e){
          return {
            status: 404
          };
        });
      };
    }
    return file;
  }());
  exports.Static.dir = (function(){
    function dir(dirname){
      return function(req, p){
        return exports.Static.file(path.join(dirname, p))();
      };
    }
    return dir;
  }());
  exports.Redirect = (function(){
    function Redirect(location){
      return {
        status: 302,
        headers: {
          location: location
        }
      };
    }
    return Redirect;
  }());
}).call(this);
